netsuite_altavia_erp_adapter/src/main/java/org/cardanofoundation/lob/app/netsuite_altavia_erp_adapter/service/internal/ExtractionParametersFilteringService.java

```diff
@@ -45,7 +45,7 @@ public Set<Transaction> applyExtractionParameters(Set<Transaction> txs,
                 .filter(tx -> {
                     val txTypes = userExtractionParameters.getTransactionTypes();
 
-                    return txTypes.isEmpty() || txTypes.contains(tx.getTransactionType());
+                    return null == txTypes || txTypes.isEmpty() || txTypes.contains(tx.getTransactionType());
```

Comment:
In which scenario txTypes could be null? Do we need this code or it is just extra protection but it will never happen?

Reply:
```
package org.cardanofoundation.lob.app.accounting_reporting_core.domain.core;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDate;
import java.util.List;

@Getter
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class UserExtractionParameters {

    @NotBlank
    private String organisationId;

    @Builder.Default
    private List<TransactionType> transactionTypes = List.of();

    @NotNull
    private LocalDate from;

    @NotNull
    private LocalDate to;

    @Builder.Default
    private List<String> transactionNumbers = List.of();

}
```

so actually it looks like it should always be an empty list?

Hm...